peps/pep-0771.rst

```diff
@@ -350,11 +440,258 @@ This could be achieved with e.g:
+resolving the dependency tree, but the point here is to illustrate that there
+may be changes in behavior in packaging tools due to the change in the
+assumption about what impact an extra can have.
+
+Security Implications
+=====================
+
+There are no known security implications for this PEP.
+
+How to teach this
+=================
+
+Package authors
+---------------
+
+While the mechanism used to define extras and the associated rule about when to
+use it are simple, package authors need to carefully consider several points
+before adopting this capability in their packages, as this has implications for
+backward-compatibility.
+
+Supporting older versions of package installers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Package installers such as `pip`_ or `uv <https://docs.astral.sh/uv/>`_ will not
+necessarily implement support for default extras at the same time, and once they
+do it is likely that package authors will want to keep supporting users who do
+not have the most recent version of a package installer. In this case, the
+following recommendations would apply:
+
+* Moving a package from being a required dependency to a default extra would be
+  a breaking change, because older versions of package installers would not
+  recognise the concept of default extras, and would then install the package
+  with fewer dependencies, which could affect users that would have been relying
+  on these. Therefore, changing dependencies from required to a default extra in
+  an established package should only be done in future once the developers only
+  want to support users with installers that implement this PEP.
+
+* Making an existing extra become a default should be safer, such as making
+  ``recommended`` in `astropy`_ be a default extra, but in order to support users
+  with older versions of package installers, the documentation should still mention
+  the extra explicitly as long as possible (until it is clear that most/all users
+  are using package installers that implement this PEP). There is no downside to
+  keeping the extra be explicitly mentioned, but this will ensure that users with
+  modern tooling who do not read documentation (which may be a non-negligeable
+  fraction of the user community) will start getting the recommended
+  dependencies by default.
+
+* Adding a new extra, whether it be ``minimal`` or another new extra that is to
+  be the default, comes with the same caveats that it does prior to this PEP, which
+  is that users will only be able to use this extra for releases that define
+  this extra. This might seem obvious, but consider a package that has a version
+  1.0 prior to using default extras. Suppose that package now defines
+  ``minimal`` in 2.0, then downstream users and packages that want to depend on
+  a minimal version of the package cannot declare the following dependency::
+
+    packaage[minimal]>=1.0
```

Comment:
```suggestion
    package[minimal]>=1.0
```

Reply:
Thanks! @willingc made the same suggestion and I committed hers